### ISSUE TITLE
Has and Belongs to Many

### ISSUE THREAD

**Issue by ghost** [activeadmin/inherited_resources#309]
Hey, I decided to use inherited_resources to make things a lot easier but i've run into a problem. In my app i have a :has_and_belongs_to_many association with a joining table.

Developer

```
extend FriendlyId
friendly_id :username, use: :slugged
has_and_belongs_to_many :organizations
```

Organization

```
extend FriendlyId
friendly_id :name, use: :slugged
has_and_belongs_to_many :developers
```

In my Organization controller i have:

``` ruby
class OrganizationsController < InheritedResources::Base
  before_filter :authenticate_developer!

  protected

  def begin_of_association_chain
    @current_developer
  end

  def collection
    @organizations ||= end_of_association_chain.page(params[:page] || 1).per(5)
  end

  def permitted_params
    params.permit(organization: :name)
  end
end
```

But it returns all the organizations in the whole table and doesn't add a relation to the developers_organizations joining table? Is there any fix for this?

**Comment by joelmoss** [comment]
Your controller knows nothing about any relationships that your models have, and shouldn't do.

What do you have in your view?

**Comment by ghost** [comment]
Well, In my index i have

```
- if @organizations.blank?
  .jumbotron
    %h3 You Don't Have Any Organizations!
- else
  = render 'organizations/organizations'

which loads a partial:

%table.table.table-bordered{ :id => 'organizations' }
  %thead
    %tr
      %th Name
  %tbody
    - @organizations.each do |organization|
      %tr
        %td
          %a{ :href => organization_path(organization) }= organization.name
        %td
          %a.pull-right{ :href => edit_organization_path(organization) } Edit

= paginate @organizations
```

I'm just concerned because the joining table remains blank no matter how many organizations are created.

**Comment by joelmoss** [comment]
I think you are expecting something of IR that it does not provide. When creating, IR simply accepts the params for the model and calls save. It is your model that determines if any relational data is created at that time.

**Comment by mcary** [comment]
FYI, I was looking for similar behavior.  I agree with @joelmoss in that this is not necessarily inherited_resources's concern.  Perhaps it's a missing feature in habtm associations?

I found a solution, which was to override the association's build method to link up the parent correctly.  I've published that solution alongside my failed attempts in this repository: [mcary/habtm-build](https://github.com/mcary/habtm-build).  Each branch has a different solution as the last commit.
